modules/brushingandlinking/src/brushingandlinkingmanager.cpp

```diff
@@ -363,6 +401,30 @@ void BrushingAndLinkingManager::propagate(BrushingAction action, BrushingTarget
     }
 }
 
+void BrushingAndLinkingManager::propagate(BrushingAction action,
+                                          const std::vector<BrushingTarget>& targets) {
+    for (auto t : targets) {
+        modifications_[t] |= fromAction(action);
+    }
+
+    if (std::holds_alternative<BrushingAndLinkingOutport*>(owner_)) {
+        auto outport = std::get<BrushingAndLinkingOutport*>(owner_);
+        outport->getProcessor()->invalidate(invalidationLevel_);
```

Comment:
Is there a need for locking the network during propagation?

Reply:
from a manager perspective I don't think so since it will only invalidate ports/processors.